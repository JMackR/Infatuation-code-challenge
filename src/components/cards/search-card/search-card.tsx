import React from 'react';
import {
    View,
    StyleSheet,
    Dimensions,
    Image,
    TouchableOpacity
} from 'react-native';
import { ReposTypes } from '../../../hooks';
import { SVG } from '../../svgs';
import { FavoriteIcon } from '../../../assets/svgs/image-catalog';
import { CoreCard } from '../core-card';

const { height } = Dimensions.get('window');

export interface SearchCardProps extends ReposTypes {
    saveToFavorites: (props: ReposTypes) => void
    testID: string
}

export const SearchCard = (props: SearchCardProps) => {
    const {
        saveToFavorites,
        testID,
        ...rest
    } = props;

    return (
        <TouchableOpacity style={styles.cardContainer} testID={testID} onPress={() => saveToFavorites(rest)} accessibilityLabel={`${testID}-user-container`}>
            <View style={styles.mainContainer}>
                {props.avatarUrl && <View style={styles.imageContainer}>
                    <Image
                        style={styles.image}
                        source={{
                            uri: props.avatarUrl,
                        }}
                        testID='avatar-url'
                    />
                </View>}
                <CoreCard {...props} />
                <View style={[styles.favIcon]}>
                    <SVG localSVG={FavoriteIcon} tint={props.isFav ? '#051BFC' : '#C2BABA'} testID='fav-icon' />
                </View>
            </View>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    cardContainer: {
        flex: 1,
        borderWidth: StyleSheet.hairlineWidth,
        flexDirection: 'row',
        borderRadius: 5,
        marginLeft: 20,
        marginRight: 15
    },
    imageContainer: {
        margin: 5,
        borderRadius: 10,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.95,
        shadowRadius: 4,
        elevation: 5,
        justifyContent: "center"
    },
    image: {
        height: height * 0.1,
        aspectRatio: 8 / 12,
        alignItems: "center",
        resizeMode: "contain"
    },

    mainContainer: {
        flex: 1,
        flexDirection: 'row',
    },

    favIcon: {
        marginRight: 10,
        justifyContent: "center"
    }


});
