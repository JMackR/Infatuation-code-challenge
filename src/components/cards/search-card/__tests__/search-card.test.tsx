import { render, fireEvent } from '@testing-library/react-native'
import { SearchCard, SearchCardProps } from '../search-card'
import renderer from 'react-test-renderer'


describe('Search Card is displayed', () => {
    let props: SearchCardProps
    beforeEach(() => {
        props = {
            id: "1",
            fullName: "hello jim",
            createdAt: "02/02/2023",
            stargazersCount: 10,
            language: "react",
            url: "http://bob.com",
            avatarUrl: "https://fastly.picsum.photos/id/1059/200/300.jpg?hmac=YMMfXR3NQPHd070_dI4t9aZU990UmWDRKmklFNF6xMw",
            saveToFavorites: jest.fn(),
            isFav: true,
            testID: "1"
        }
    })
    it('displays the repo image', () => {
        const { getByTestId } = render(<SearchCard {...props} />)
        expect(getByTestId('avatar-url').props.children).toMatchSnapshot();
    });


    test("check fav icon is active", () => {
        const tree = renderer.create(<SearchCard {...props} isFav={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("check fav icon is inactive", () => {
        const tree = renderer.create(<SearchCard {...props} isFav={false} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('is Pressed', async () => {
        const { getByTestId } = render(<SearchCard {...props} />)
        const press = getByTestId(props.testID as string)
        fireEvent.press(press);
        expect(props.saveToFavorites).toHaveBeenCalledTimes(1)
    })
})

