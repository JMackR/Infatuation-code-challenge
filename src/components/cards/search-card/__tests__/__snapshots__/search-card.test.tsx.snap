// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Card is displayed check fav icon is active 1`] = `
<View
  accessibilityLabel="1-user-container"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 5,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "row",
      "marginLeft": 20,
      "marginRight": 15,
      "opacity": 1,
    }
  }
  testID="1"
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "borderRadius": 10,
          "elevation": 5,
          "justifyContent": "center",
          "margin": 5,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.95,
          "shadowRadius": 4,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://fastly.picsum.photos/id/1059/200/300.jpg?hmac=YMMfXR3NQPHd070_dI4t9aZU990UmWDRKmklFNF6xMw",
          }
        }
        style={
          {
            "alignItems": "center",
            "aspectRatio": 0.6666666666666666,
            "height": 133.4,
            "resizeMode": "contain",
          }
        }
        testID="avatar-url"
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
          "marginLeft": 5,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Name:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="fullName"
        >
          hello jim
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Stargazers Count:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="stargazersCount"
        >
          10
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Language:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="language"
        >
          react
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Year Created:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="createdAt"
        >
          2023
        </Text>
      </View>
      <View
        style={
          [
            {
              "flex": 1,
              "flexDirection": "row",
              "gap": 10,
              "marginVertical": 2,
            },
          ]
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Link:
        </Text>
        <View
          style={
            [
              {
                "flex": 1,
                "marginRight": 15,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontFamily": "Lato-Regular",
                "fontSize": 12,
              }
            }
            testID="url"
          >
            http://bob.com
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
            "marginRight": 10,
          },
        ]
      }
    >
      <SvgMock
        fill="#051BFC"
      />
    </View>
  </View>
</View>
`;

exports[`Search Card is displayed check fav icon is inactive 1`] = `
<View
  accessibilityLabel="1-user-container"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderRadius": 5,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "row",
      "marginLeft": 20,
      "marginRight": 15,
      "opacity": 1,
    }
  }
  testID="1"
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "borderRadius": 10,
          "elevation": 5,
          "justifyContent": "center",
          "margin": 5,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.95,
          "shadowRadius": 4,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://fastly.picsum.photos/id/1059/200/300.jpg?hmac=YMMfXR3NQPHd070_dI4t9aZU990UmWDRKmklFNF6xMw",
          }
        }
        style={
          {
            "alignItems": "center",
            "aspectRatio": 0.6666666666666666,
            "height": 133.4,
            "resizeMode": "contain",
          }
        }
        testID="avatar-url"
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
          "marginLeft": 5,
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Name:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="fullName"
        >
          hello jim
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Stargazers Count:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="stargazersCount"
        >
          10
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Language:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="language"
        >
          react
        </Text>
      </View>
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 10,
            "marginVertical": 2,
          }
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Year Created:
        </Text>
        <Text
          style={
            {
              "fontFamily": "Lato-Regular",
              "fontSize": 12,
            }
          }
          testID="createdAt"
        >
          2023
        </Text>
      </View>
      <View
        style={
          [
            {
              "flex": 1,
              "flexDirection": "row",
              "gap": 10,
              "marginVertical": 2,
            },
          ]
        }
      >
        <Text
          style={
            {
              "flexWrap": "wrap",
              "fontFamily": "Lato-Bold",
              "fontSize": 12,
            }
          }
        >
          Link:
        </Text>
        <View
          style={
            [
              {
                "flex": 1,
                "marginRight": 15,
              },
            ]
          }
        >
          <Text
            style={
              {
                "fontFamily": "Lato-Regular",
                "fontSize": 12,
              }
            }
            testID="url"
          >
            http://bob.com
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
            "marginRight": 10,
          },
        ]
      }
    >
      <SvgMock
        fill="#C2BABA"
      />
    </View>
  </View>
</View>
`;

exports[`Search Card is displayed displays the repo image 1`] = `undefined`;
